src/governance.cpp

```diff
@@ -954,23 +954,22 @@ void CGovernanceManager::RequestGovernanceObject(CNode* pfrom, const uint256& nH
         return;
     }
 
-    CBloomFilter filter(GOVERNANCE_FILTER_ELEMENTS, GOVERNANCE_FILTER_FP_RATE, GetRandInt(999999), BLOOM_UPDATE_ALL);
+    CBloomFilter filter;
+    filter.clear();
 
     if(fUseFilter) {
         CGovernanceObject* pObj = FindGovernanceObject(nHash);
 
         if(pObj) {
+            int nFilterElements = min(3*mnodeman.CountMasternodes(), GOVERNANCE_FILTER_MAX_ELEMENTS); 
```

Comment:
Why? Testnet obj/vote ratio is reverted comparing to mainnet - much more objs, much less votes per obj on testnet. So you'll end up pushing 50MB of bloom filters on testnet for the same amount of votes which would require 1MB of bloom filters on mainnet. By having net-dependent size you cam simulate comparable net load generated by bloom for the same total number of votes.